kernel/src/net/socket/ip/datagram/mod.rs

```diff
@@ -325,7 +325,9 @@ impl Socket for DatagramSocket {
         flags: SendRecvFlags,
     ) -> Result<usize> {
         // TODO: Deal with flags
-        debug_assert!(flags.is_all_supported());
+        if !flags.is_all_supported() {
+            todo!()
+        }
```

Comment:
I think this can have better messages? (And TODO comments are duplicate if we already have the `todo!` macro, so they can be removed).

```diff
diff --git a/kernel/src/net/socket/ip/stream/mod.rs b/kernel/src/net/socket/ip/stream/mod.rs
index 2cef1cf3..5b6f427d 100644
--- a/kernel/src/net/socket/ip/stream/mod.rs
+++ b/kernel/src/net/socket/ip/stream/mod.rs
@@ -565,8 +565,9 @@ impl Socket for StreamSocket {
         message_header: MessageHeader,
         flags: SendRecvFlags,
     ) -> Result<usize> {
-        // TODO: Deal with flags
-        debug_assert!(flags.is_all_supported());
+        if !flags.is_all_supported() {
+            todo!("unsupported flags: {:?}", flags);
+        }
 
         let MessageHeader {
             control_message, ..

```